### PR TITLE
Enable quick access to notifications from Reader

### LINKED ISSUE #23717: iPad: Notifications bell appears only on Home
(UX issues, based on feedback)

It's unexpected that the "Notifications" bells only appears on the Home Screen.

### PR DESCRIPTION
**Pull request by kean**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/23717 by adding the "Notifications" bell directly to the main Reader screens. If the app re-opens on one of these screens thanks to state restoration, you will see the bell regardless of whether the menu is opened or the orientation. The bell provides instant information: do I have any new notifications? It also allows you to glance over the newest notifications quickly.

Screenshot 2024-12-10 at 2 28 55 PM

To test:

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Reader/Controllers/ReaderDiscoverViewController.swift [hunk 1]
```diff
@@ -42,8 +44,29 @@ class ReaderDiscoverViewController: UIViewController, ReaderDiscoverHeaderViewDe
...
+
     private func setupNavigation() {
         navigationItem.largeTitleDisplayMode = .never
+        setupNotificationsBarButtonItem()
+    }
+
+    private func setupNotificationsBarButtonItem() {
+        notificationsButtonCancellable = nil
+        if traitCollection.horizontalSizeClass == .regular {
+            notificationsButtonCancellable =         notificationsButtonViewModel.$image.sink { [weak self] in
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/Controllers/ReaderDiscoverViewController.swift]: There are a few whitespaces after `=`.

File: WordPress/Classes/ViewRelated/Reader/Controllers/ReaderStreamViewController.swift [hunk 2]
```diff
@@ -350,13 +353,28 @@ import AutomatticTracks
...
 
     private func didChangeIsCompact(_ isCompact: Bool) {
         (tableView.tableHeaderView as? ReaderBaseHeaderView)?.isCompact = isCompact
         tableView.reloadData()
     }
 
+    private func setupNotificationsBarButtonItem() {
+        notificationsButtonCancellable = nil
+        if isNotificationsBarButtonEnabled && traitCollection.horizontalSizeClass == .regular {
+            notificationsButtonCancellable =         notificationsButtonViewModel.$image.sink { [weak self] in
```
Comment (crazytonyli) [inline WordPress/Classes/ViewRelated/Reader/Controllers/ReaderStreamViewController.swift]: Whitespaces here too.

**Review by crazytonyli — APPROVED** [review]

File: WordPress/Classes/ViewRelated/Reader/Controllers/ReaderDiscoverViewController.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Reader/Controllers/ReaderDiscoverViewController.swift]: Thanks; didn't notice it with the way my editor is set up. Fixed.

**Pull request merged**